vcd/resource_vcd_vapp_vm.go

```diff
@@ -1006,41 +1113,33 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 			}
 		}
 
-		if d.HasChange("hardware_version") {
+		if d.HasChange("hardware_version") || d.HasChange("os_type") || d.HasChange("description") {
```

Comment:
Please add a comment above this explaining the logic behind this whole block, like why we are grouping these three parameters.

Reply:
added